pkg/securitycontext/provider.go

```diff
@@ -48,9 +49,18 @@ func (p SimpleSecurityContextProvider) ModifyContainerConfig(pod *api.Pod, conta
 // ModifyHostConfig is called before the Docker runContainer call.
 // The security context provider can make changes to the HostConfig, affecting
 // security options, whether the container is privileged, volume binds, etc.
-// An error is returned if it's not possible to secure the container as requested
-// with a security context.
 func (p SimpleSecurityContextProvider) ModifyHostConfig(pod *api.Pod, container *api.Container, hostConfig *docker.HostConfig) {
+	// Apply pod security context
+	if pod.Spec.SecurityContext != nil {
+		if pod.Spec.SecurityContext.SupplementalGroups != nil && container.Name != leaky.PodInfraContainerName {
```

Comment:
@pweil- The reason to skip the infra container is the bug fixed by opencontainers/runc#313, where a container without an /etc/groups file can't be started if it has supplemental groups.

I _think_ we should be able to back that hack out, but for now, it should really be documented with a comment in the code.

@swagiaal